src/tests/ftest/launch.py

```diff
@@ -925,6 +933,142 @@ def archive_logs(avocado_logs_dir, test_yaml, args):
     spawn_commands(host_list, "; ".join(commands), 900)
 
 
+def human_to_bytes(h_size):
+    """Convert human readable size values to respective byte value.
+
+    Args:
+        h_size (str): human readable size value to be converted.
+    """
+    if h_size.endswith("KB") or h_size.endswith("kb"):
+        b_size = int(re.sub('[^0-9]', '', h_size)) * (2**10)
+    elif h_size.endswith("MB") or h_size.endswith("mb"):
+        b_size = int(re.sub('[^0-9]', '', h_size)) * (2**20)
+    elif h_size.endswith("GB") or h_size.endswith("gb"):
+        b_size = int(re.sub('[^0-9]', '', h_size)) * (2**30)
+    elif h_size.endswith("B") or h_size.endswith("b"):
+        b_size = int(re.sub('[^0-9]', '', h_size))
+    return b_size
+
+
+def send_notification(hosts, subject, msg, attachment, from_addr, to_addrs):
```

Comment:
Have our test software being able to send e-mails directly is not a good idea.

All Emails from CI should only be generated by Jenkins.

Which means that the test would need to be killed and either unitTestPost.groovy or functuionalTestPost.groovy would need to be changed to detect this and send the e-mail.  Jenkins already knows how to look up the recipients for a run.

And you don't want to have tests running manually to attempt to send e-mails out.

Reply:
That's a good point, when running manually we can just have the log_size.log file stored in the avocado test folders and the user can access it if needed. Would there be any way to tell Jenkins to send an email to the author of the PR without having to fail the test? If so, what would need to happen? Could we add a step on the post-step to check the log_sizes.log file in the artifacts and send the email based on the contents of the log_sizes.log? This would mean that the threshold and notifications would be handled by Jenkins and launch.py would be responsible for collecting the information in the files.

Reply:
Not with our current configuration.  Currently Jenkins just runs the test as a shell script and waits for it to complete.
And we can wrap that step with a timeout to kill the step if it is too long.

There are some plugins that claim to add some ability to run parallel actions, but I do not know if they would work for that type of monitoring.   We do not currently install those plugins.

Reply:
@amjustin13 I didn't realise/think the goal here was to alert developers (yet), but thought it was to gather information on existing log sizes so that we can decide what the _high water_ thresholds were for flagging a log as being **too big**.

When it comes time to alerting developers, @JohnMalmberg  is correct that sending e-mails is not the correct mechanism.  Failing the test, or more better yes, creating a new _failing_ junit result file that reports the test, and log file sizes is the correct mechanism.  That way the developer will see the reason for the failure in the _Tests_ tab (or Test Results link on the classic UI) along with any other test results.